pypesto/hierarchical/optimal_scaling_approach/optimal_scaling_calculator.py

```diff
@@ -0,0 +1,170 @@
+import copy
+from typing import Dict, List, Sequence, Tuple
+
+import numpy as np
+
+from ...C import FVAL, GRAD, HESS, MODE_RES, RDATAS, RES, SRES, X_INNER_OPT
+from ...objective.amici.amici_calculator import AmiciModel, AmiciSolver
+from ...objective.amici.amici_util import (
+    filter_return_dict,
+    init_return_values,
+)
+from .optimal_scaling_problem import OptimalScalingProblem
+from .optimal_scaling_solver import OptimalScalingInnerSolver
+
+try:
+    import amici
+    from amici.parameter_mapping import ParameterMapping
+except ImportError:
+    pass
+
+
+class OptimalScalingAmiciCalculator:
+    """A calculator is passed as `calculator` to the pypesto.AmiciObjective.
+
+    While this class cannot be used directly, it has two subclasses
```

Comment:
This does not seem to be the case: (1) this class has no abstract methods, (2) there seem to be no forward/adjoint subclasses. Legacy comment?